godel_process_planning/src/blend_process_planning.cpp

```diff
@@ -0,0 +1,157 @@
+#include <godel_process_planning/godel_process_planning.h>
+
+#include <ros/console.h>
+
+// descartes
+#include "descartes_trajectory/axial_symmetric_pt.h"
+#include "descartes_trajectory/joint_trajectory_pt.h"
+#include "descartes_planner/dense_planner.h"
+
+#include "common_utils.h"
+
+
+namespace godel_process_planning
+{
+
+/**
+ * @brief toDescartesPt
+ * @param pose
+ * @return
+ */
+static inline descartes_core::TrajectoryPtPtr toDescartesPt(const Eigen::Affine3d& pose, double dt)
+{
+  using namespace descartes_trajectory;
+  using namespace descartes_core;
+  const descartes_core::TimingConstraint tm (dt);
+  return TrajectoryPtPtr(new AxialSymmetricPt(pose, M_PI/12.0, AxialSymmetricPt::Z_AXIS, tm)); // default timing
+}
+
+static inline double costFunction(const std::vector<double>& source,
+                                  const std::vector<double>& target)
+{
+  double cost = 0.0;
+  for (std::size_t i = 0; i < source.size(); ++i)
+  {
+    double diff = std::abs(source[i] - target[i]);
+    if (diff > M_PI_2) cost += 5.0 * diff;
+    else cost += diff;
+  }
+  return cost;
+}
+
+/**
+ * @brief toDescartesTraj
+ * @param ref
+ * @param points
+ * @return
+ */
+static godel_process_planning::DescartesTraj toDescartesTraj(const geometry_msgs::Pose& ref,
+                                                             const std::vector<geometry_msgs::Point>& points,
+                                                             const godel_msgs::BlendingPlanParameters& params)
+{
+  DescartesTraj traj;
+  traj.reserve(points.size());
+  if (points.empty()) return traj;
+
+  Eigen::Affine3d last_pose = createNominalTransform(ref, points.front());
+
+  for (std::size_t i = 0; i < points.size(); ++i)
+  {
+    Eigen::Affine3d this_pose = createNominalTransform(ref, points[i]);
+    double dt = (this_pose.translation() - last_pose.translation()).norm() / params.traverse_spd;
+    traj.push_back( toDescartesPt(this_pose, dt) );
+    last_pose = this_pose;
+  }
+
+  return traj;
+}
+
+bool ProcessPlanningManager::handleBlendPlanning(godel_msgs::BlendProcessPlanning::Request& req,
+                                                 godel_msgs::BlendProcessPlanning::Response& res)
+{
+  // Transform process path from geometry msgs to descartes points
+  DescartesTraj process_points = toDescartesTraj(req.path.reference, req.path.points, req.params);
+
+  // Capture the current state of the robot
+  std::vector<double> current_joints = getCurrentJointState("joint_states");
```

Comment:
I don't link that this node has this kind of state associated with it. It would be best if this was part of the request. I don't know that I'll change this now, however.